src/LabelUtil/LabelUtil.js

```diff
@@ -0,0 +1,85 @@
+import select from 'd3-selection/src/select';
+
+/**
+ * Utilities to word wrap labels automatically.
+ */
+class LabelUtil {
+
+  /**
+   * Determines if labels can be wrapped
+   *
+   * @param  {d3.selection} root the root node to search for text from
+   * @param  {String} subSelector The subSelector for single text elements
+   * @param  {Integer} leftPadding The amount of padding to the left,
+   *     beginning from the second line. Needed for legends with images
+   * @param  {Integer} lineHeight The lineHeight in em to use
+   * @param {Boolean} ignoreNeighbors Flag indicating if neighbors should
+   *     be respected. Should be false e.g. when used on an x-axis.
+   * @param {Integer} distanceBuffer The buffer used for spacing between labels
+   */
+  static handleLabelWrap(root, subSelector, leftPadding, lineHeight, ignoreNeighbors, distanceBuffer) {
+    const node = root.node();
+    if (!node) {
+      return;
+    }
+    const width = node.getBoundingClientRect().width;
+    let length;
+    if (ignoreNeighbors) {
+      length = width - distanceBuffer - (leftPadding || 0);
+    } else {
+      const neighborCount = root.selectAll(subSelector || ' g > text').size();
+      length = width / neighborCount - distanceBuffer - (leftPadding || 0);
+    }
+    root.selectAll(subSelector || ' g > text')
+      .call(this.wordWrap, length, leftPadding, lineHeight);
+  }
+
+  /**
+   * Handles word wrapping
+   *
+   * @param  {Object} textEl The text element collection
+   * @param  {Integer} width The maximum width to use
+   * @param  {Integer} leftPadding The amount of padding to the left
+   * @param  {Integer} lineHeight The lineHeight in em to use
+   */
+  static wordWrap(textEl, width, leftPadding, lineHeight) {
+    textEl.each(function() {
```

Comment:
see above comment

Reply:
The worst that could happen is that there aren't any label nodes, so we'd iterate over an empty selection here.